components/authz-service/storage/postgres/migration/sql/47_staged_rules_tables.up.sql

```diff
@@ -0,0 +1,94 @@
+BEGIN;
+
+CREATE TABLE iam_staged_projects (
+  db_id SERIAL,
+  id TEXT PRIMARY KEY,
+  name TEXT NOT NULL,
+  type iam_policy_type NOT NULL DEFAULT 'custom',
+  projects TEXT[],
+  deleted BOOLEAN,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+);
+
+CREATE TABLE iam_staged_project_rules (
+  db_id SERIAL PRIMARY KEY,
+  id TEXT NOT NULL UNIQUE,
+  project_id TEXT REFERENCES iam_staged_projects ON DELETE CASCADE,
+  name TEXT NOT NULL,
+  type TEXT NOT NULL,
+  deleted BOOLEAN,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+);
+
+CREATE TABLE iam_staged_rule_conditions (
+  db_id SERIAL PRIMARY KEY,
+  rule_db_id INTEGER REFERENCES iam_staged_project_rules ON DELETE CASCADE,
+  value TEXT[] NOT NULL,
+  attribute TEXT NOT NULL,
+  operator TEXT NOT NULL,
+  deleted BOOLEAN,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE OR REPLACE FUNCTION
+  projects_match_for_staged_rule(_project_id TEXT, _projects_filter TEXT[])
+  RETURNS BOOLEAN AS $$
+    BEGIN
+      RETURN (
+        array_length(_projects_filter, 1) IS NULL
+        OR _project_id = ANY (_projects_filter)
+      );
+    END
+$$ LANGUAGE PLPGSQL;
```

Comment:
Not seeing how these functions are different besides the name:

https://github.com/chef/automate/blob/master/components/authz-service/storage/postgres/migration/sql/44_policy_rules.up.sql#L19-L28

Reply:
Oh yep -- missing the corresponding `query_staged_rule`. I suppose no need for the rename but maybe rename both for clarity 🤔 . Not sure yet how much of the db work around "current" rules will need to stay since we'll mostly be using/displaying/manipulating staged data.